ce/api/geo.py

```diff
@@ -1,6 +1,101 @@
+import logging
+import collections
+import sys
+from collections import OrderedDict
+from threading import RLock
```

Comment:
If we use a multiprocess (rather than multithreaded) approach to handling requests, do we even need to lock the cache?